siriuspy/siriuspy/optics/lattice_survey.py

```diff
@@ -31,59 +32,103 @@ def generate_model_static_table():
 
     model = pymodels.bo
     acc = model.create_accelerator()
     ind = pyaccel.lattice.find_indices(acc, 'fam_name', 'InjSept')
     acc = pyaccel.lattice.shift(acc, ind[0])
-    _append_mag_data(filename, model, acc, 'Booster', 'BO')
+    _append_mag_data(filename, model, acc, 'Booster (BO)', 'BO')
 
     model = pymodels.tb
-    acc, *_ = model.create_accelerator()
-    _append_mag_data(filename, model, acc, 'Booster Transport Line', 'TB')
+    acc, *_ = model.create_accelerator(add_from_li_triplets=False)
+    _append_mag_data(
+        filename, model, acc, 'Booster Transport Line (TB)', 'TB')
 
     model = pymodels.ts
     acc, *_ = model.create_accelerator()
-    _append_mag_data(filename, model, acc, 'SR Transport Line', 'TS')
+    _append_mag_data(filename, model, acc, 'SR Transport Line (TS)', 'TS')
+
+
+def _conv_inst(section, sub, mag_tp, inst):
+    inst_conv = {
+        'TS-01': {
+            'CV': {'2': '1E2', '3': '2'},
+            },
+        'TS-02': {
+            'CV': {'1': '0', '2': ''},
+            },
+        'TS-04': {
+            'CV': {'1': '0', '2': '1', '3': '1E2', '4': '2'},
+            },
+        }
+    secsub = section + '-' + sub
+    inst_conv_ = inst_conv
+    if secsub in inst_conv_:
+        inst_conv_ = inst_conv_[secsub]
+        if mag_tp in inst_conv_:
+            inst_conv_ = inst_conv_[mag_tp]
+            if inst in inst_conv_:
+                return inst_conv_[inst]
+    return inst
 
 
 def _append_mag_data(filename, model, acc, label, section):
     fam_data = model.get_family_data(acc)
     pos = pyaccel.lattice.find_spos(acc)
 
+    args = (acc, ) if model == pymodels.tb else ()
     mag_tps = []
-    mag_tps.extend(model.families.families_dipoles())
-    mag_tps.extend(model.families.families_pulsed_magnets())
-    mag_tps.extend(model.families.families_quadrupoles())
-    mag_tps.extend(model.families.families_horizontal_correctors())
-    mag_tps.extend(model.families.families_vertical_correctors())
-    mag_tps.extend(model.families.families_sextupoles())
-    mag_tps.extend(model.families.families_skew_correctors())
-    if section.lower().startswith('si'):
-        mag_tps.extend(model.families.families_id_correctors())
+    mag_tps.extend(model.families.families_dipoles(*args))
+    mag_tps.extend(model.families.families_pulsed_magnets(*args))
+    mag_tps.extend(model.families.families_quadrupoles(*args))
+    mag_tps.extend(model.families.families_horizontal_correctors(*args))
+    mag_tps.extend(model.families.families_vertical_correctors(*args))
+    mag_tps.extend(model.families.families_sextupoles(*args))
+    mag_tps.extend(model.families.families_skew_correctors(*args))
 
     mag_data = dict()
     for mag_tp in mag_tps:
         inds = fam_data[mag_tp]['index']
         subs = fam_data[mag_tp]['subsection']
         insts = fam_data[mag_tp]['instance']
         for ind, inst, sub in zip(inds, insts, subs):
+            # --- ID correctors
```

Comment:
@xresende, I think it is better to create a method to make these conversions, don't you think?

Reply:
yes, done.